File: polyglot/Primes/primes.hs
import System.Environment -- getArgs
import Data.List  -- intercalate

sieve :: [Int] -> [Int]
sieve (x:xs) = x:sieve [y | y <- xs, mod y x /= 0]

primes :: [Int]
primes = sieve [2..]

main :: IO ()
main = do
  [arg] <- getArgs
  let numPrimes = read arg ::Int in
    putStrLn $ "(" ++ 
                intercalate " " (map show (take numPrimes primes))
                ++ ")"

File: polyglot/Primes/primes.clj
(ns primes (:gen-class))

(defn sieve [s]
  (let [x (first s)]
    (cons x (lazy-seq (sieve (filter #(not= 0 (mod % x)) (rest s)))))))

(defn -main [arg]
  (def numPrimes (read-string arg))
  (def from2 (iterate inc 2))
  (def primes (sieve from2))
  (println (take numPrimes primes)))


;(-main "138")
